share/completions/rustc.fish

```diff
@@ -49,13 +49,13 @@ for line in $rust_docs
 end
 
 set -l rust_docs (rustc -W help  \
-    | egrep \
-        '(\s+)(.+)(\s+)(allow|warn|deny|forbid)(\s+){2}([^\n]+)' \
+    | string match -r \
```

Comment:
I feel like there should be somethin way with just string replace and the pcre2 syntax if  better understood regexps I could do this with one string replace... I can't nail it.